.buildkite/scripts/commands.py

```diff
@@ -4,27 +4,36 @@
 import sys
 
 
-def check_call(cmd, exit_on_error=True):
-    """
-    A wrapped version of subprocess.check_call() that doesn't print a
-    traceback if the command errors.
-    """
+def _subprocess_run(cmd, exit_on_error=True):
     print("*** Running %r" % " ".join(cmd))
-    try:
-        return subprocess.check_output(cmd).decode("utf8").strip()
-    except subprocess.CalledProcessError as err:
-        if exit_on_error:
-            sys.exit(err.returncode)
-        pass
+
+    output = []
+    pipe = subprocess.Popen(
+        cmd, encoding="utf8", stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+    )
+
+    # Await command completion and print lines as they come in
+    for stdout_line in iter(pipe.stdout.readline, ""):
+        print(stdout_line.strip())
+        output.append(stdout_line)
+
+    # Extract results
+    pipe.communicate()
+    return_code = pipe.returncode
+
+    if return_code != 0 and exit_on_error:
+        sys.exit(return_code)
+
+    return "\n".join(output).strip()
```

Comment:
If you do stick with this, you want `"".join(output)` instead of `"\n".join(output)`; the latter is inserting an extra newline between consecutive lines.